aws/resource_aws_launch_configuration.go

```diff
@@ -99,7 +99,7 @@ func resourceAwsLaunchConfiguration() *schema.Resource {
 						return ""
 					}
 				},
-				ValidateFunc: validateInstanceUserDataSize,
+				ValidateFunc: validateMaxLength(16384),
```

Comment:
I feel like this may be problematic if unicode is used (not that it was correct before). If the user data limit is on the actual encoded byte length, not the string character length, it could be exhausted before this limit is hit (sorry I didn't catch this the first time around). Can you confirm if it is actually bytes like the previous error message said, and if so, do you mind possibly adding a validation for an encoded byte length instead of just string length?

Reply:
Nevermind, I was confused between byte length from `len` and rune counts. Assuming user data is in UTF-8, this will be fine.